ImportExport/CadbaseLibrary.FCMacro

```diff
@@ -0,0 +1,643 @@
+- then you will be prompted to select a folder for data (where the files from the CADBase storage will be loaded)
+A widget "CADBase Library" will be generated in a separate window.
+- choose tab "Options" in window of the macro
+- click "Settings" button
+- in window that opens, you need to set the username and password to gain access to CADBase
+- after press the "OK" button and wait for the token to be received
+
+USAGE
+
+- for update access token need in the "CADBase library" window, in the "Options" tab, click the "Settings" button, in the "CADBase library configuration" window that opens, you need to set the username/password to gain access to CADBase and wait for a new token to be received after pressing the "OK" button
+- add target components to bookmarks (favorites) on the CADBase site. In FreeCAD will only display components that the user has bookmarked on CADBase, as well as those that have been previously downloaded
+- clicking "Update from CADBase" only updates the list of components from bookmarks active user, without downloading component modifications and files
+- double-clicking on a component folder to get component's modifications
+- getting files of a fileset for FreeCAD occurs after double-clicking on the modification folder
+- double-clicking on file for use it
+
+COMMUNICATION
+
+If you want to share a tip, get involved, report a problem, or anything else, please create a issue or post in this FreeCAD forum thread:
+https://forum.freecadweb.org/viewtopic.php?f=22&t=69389
+'''
+
+from __future__ import print_function
+
+__Title__= 'CADBase Library'
+__Name__ = 'CadbaseLibrary'
+__Comment__ = 'This CadbaseLibrary macro to use components (parts) from CADBase in app.'
+__Author__ = 'mnnxp'
+__Date__ = '2022-06-13'
+__Version__ = '0.1.0'
+__License__ = 'LGPL-3.0-or-later'
+__Web__ = 'http://www.freecadweb.org/wiki/Macro_CadbaseLibrary'
+__Wiki__ = 'http://www.freecadweb.org/wiki/Macro_CadbaseLibrary'
+__Icon__ = 'CadbaseLibrary.svg'
+__Help__ = 'Requires CADBase account to correct use the macro.'
+__Status__ = 'Stable'
+__Requires__ = 'FreeCAD >= 0.19'
+__Communication__ = 'https://forum.freecadweb.org/viewtopic.php?f=22&t=69389'
+__Files__ = 'CadbaseLibrary/Macro_CadbaseLibrary.md,CadbaseLibrary/cadbase_library.ui,CadbaseLibrary/cadbase_library_config.ui,CadbaseLibrary.svg'
+
+import os
+import sys
+import time
+import zipfile
+import tempfile
+import subprocess
+import json
+import pathlib
+from multiprocessing import cpu_count
+from multiprocessing.pool import ThreadPool
+from types import SimpleNamespace
+from PySide import QtGui, QtCore
+from PySide2 import QtNetwork
+import Part
+import Mesh
+import FreeCAD as app
+import FreeCADGui as gui
+
+param = app.ParamGet('User parameter:Plugins/cadbase_library')
+destination = param.GetString('destination')
+api = param.GetString('api-url')
+token = param.GetString('auth-token')
+g_api_login = 'https://api.cadbase.rs/login'
+g_cdbs_api = "https://api.cadbase.rs/graphql"
```

Comment:
I'd be better to be consistent in your usage of single or double quotes but this is not fundamental.